File: solutions/_1160_find_words_that_can_be_formed_by_characters/solution.py
from typing import List


class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        def count(word):
            for c in word:
                if word.count(c) > chars.count(c):
                    return 0
            return len(word)

        return sum(count(word) for word in words)


# https://leetcode.com/problems/find-words-that-can-be-formed-by-characters/

File: solutions/_1160_find_words_that_can_be_formed_by_characters/solution.js
/**
 * @param {string[]} words
 * @param {string} chars
 * @return {number}
 */
var countCharacters = function (words, chars) {
  return words.reduce((acc, word) => {
    for (const ch of word) {
      if (count(word, ch) > count(chars, ch)) {
        return acc;
      }
    }
    return acc + word.length;
  }, 0);
};

function count(word, ch) {
  let count = 0;
  for (let i = 0; i < word.length; i++) {
    if (word[i] == ch) {
      count++;
    }
  }
  return count;
}

module.exports = { countCharacters };

/*
https://leetcode.com/problems/find-words-that-can-be-formed-by-characters/
*/
